pkg/beacon/relay/groupselection/ticket.go

```diff
@@ -79,6 +80,11 @@ func calculateTicketValue(
 	return keccak256Hash, nil
 }
 
+// Returns 8 starting bytes of ticket value as a big integer.
```

Comment:
No, wait. You are right here. I need to go through this code one more time.

Reply:
This is very misleading. On-chain we speak about value as the first 8 bytes. Off-chain, value is all 32 bytes. This code is right, the naming is wrong. But it's not something for this PR.

A kind request: can we copy the on-chain explanation of value bytes to `ticket` struct to avoid further confusion?

```
type ticket struct {
  // first 8 bytes: result of keccak256 cryptography hash
  // function on the combination of the group selection seed (previous
  // beacon output), staker-specific value (address) and virtual staker index.
  // next 20 bytes: a staker-specific value which is the address of the staker.
  // next 4 bytes: virtual staker index; a number within a range of 1 to staker's weight;
  value [32]byte // W_k

  proof *proof   // proof_k = Proof(Q_j, vs)
}
```

Reply:
Sure, but the comment is not 100% right with the actual meaning of first `8` bytes, I think it should be:
```
type ticket struct {
	// Result of keccak256 cryptographic hash function on the combination of:
	// - group selection seed (previous beacon output)
	// - staker-specific value (address)
	// - virtual staker index;
	// From the chain's perspective it is composed of:
	// - first 8 bytes: ticket value
	// - next 20 bytes: a staker-specific value which is the address of the staker
	// - next 4 bytes: virtual staker index; a number within a range of 1 to staker's weight;
	value [32]byte // W_k

	proof *proof   // proof_k = Proof(Q_j, vs)
}
```

Reply:
Nope. My first comment was correct. Please look at `packTicket` function in `ethereum.go`. We should be fine with 8-bytes value here.

Reply:
Yeah but in this case we must change `ticket` struct and their `value` should hold only the first `8` bytes of the hash returned by `calculateTicketValue`. Let's address this problem in a separate pull request.